src/libraries/System.Private.CoreLib/src/System/Collections/BitArray.cs

```diff
@@ -943,6 +943,55 @@ public bool HasAnySet()
             return (_array[byteCount] & mask) != 0;
         }
 
+        /// <summary>Computes the number of bits that are set in the <see cref="BitArray"/>.</summary>
+        /// <returns>The number of set bits in the <see cref="BitArray"/>.</returns>
+        public int PopCount()
+        {
+            int count = 0;
+            foreach (int i in MemoryMarshal.Cast<byte, int>(_array))
+            {
+                count += int.PopCount(i);
+            }
+
+            return count;
+        }
+
+        /// <summary>Computes the number of trailing zero bits in the <see cref="BitArray"/>.</summary>
+        /// <returns>The number of trailing zero bits in the <see cref="BitArray"/>.</returns>
+        public int TrailingZeroCount()
+        {
+            int count = 0;
+            foreach (byte b in _array)
+            {
+                if (b != 0)
+                {
+                    return byte.TrailingZeroCount(b) + count;
+                }
+
+                count += BitsPerByte;
+            }
+
+            return _bitLength;
+        }
+
+        /// <summary>Computes the number of leading zero bits in the <see cref="BitArray"/>.</summary>
+        /// <returns>The number of leading zero bits in the <see cref="BitArray"/>.</returns>
+        public int LeadingZeroCount()
+        {
+            int count = _bitLength - _array.Length * BitsPerByte; // offset for extra bits beyond _bitLength
+            for (int i = _array.Length - 1; i >= 0; i--)
+            {
+                if (_array[i] != 0)
+                {
+                    return byte.LeadingZeroCount(_array[i]) + count;
```

Comment:
This is exposing 4-byte storage alignment that is an internal implementation detail. Are there any other APIs on BItArray leak this internal implementation detail?

I would expect the leading zero-bit count to be leading zero-bits count in the most significant byte, and ignore any internal storge alignment.

(The concept of leading zero-bit count is a bit question for a variable sized "number".)